File: String/C++/reverseStr.cpp
#include <iostream>
#include <string>
using namespace std;

string sol(string str) {
    if (str.empty()) {
        return str;
    }
    return sol(str.substr(1)) + str[0];
}

int main() {
    cout << sol("Hello") << endl;
    return 0;
}

File: String/Java/reverseStr.java
public class reverseStr {
    public static String sol(String str) {
        if (str.isEmpty()) {
            return str;
        }
        return sol(str.substring(1)) + str.charAt(0);
    }

    public static void main(String[] args) {
        System.out.println(sol("Hello"));
    }
}